### PR TITLE
make 'audio' and 'stream' independent of 'call'

### PR DESCRIPTION
**Pull request by jmillan**
With these few changes baresip `audio` and `stream` do not depend on `call`

### CODE REVIEW

File: src/stream.c [hunk 1]
```diff
@@ -383,7 +383,7 @@ int stream_alloc(struct stream **sp, const struct stream_param *prm,
 	s->rtcp  = s->cfg.rtcp_enable;
 
 	if (prm->use_rtp) {
-		err = stream_sock_alloc(s, call_af(call));
+		err = stream_sock_alloc(s, call ? call_af(call) : AF_INET);
```
Comment (jmillan) [inline src/stream.c]: I know this is limited to use `AF_INET` family when `call` is not used, yet I find it a good trade-off.

**Comment by alfredh** [comment]
thanks @jmillan this is a good improvement.

the best would be to make `stream` and `audio` and `video` compile-time
independent of `call`.

**Comment by jmillan** [comment]
> I guess that this also applies to video_alloc().

Indeed.

**Comment by jmillan** [comment]
> the best would be to make stream and audio and video compile-time
independent of call.

One may want to use both kind of streams :-)

**Comment by alfredh** [comment]
hi @jmillan 

I added address family and cname to the struct stream_param so that they
can be transported directly from call to stream:

```c
/** Common parameters for media stream */
struct stream_param {
	bool use_rtp;
	int af;
	const char *cname;
};
```

can you please rebase your PR on top of master ?

some of the changes in your PR are no longer necessary. perhaps the only
change you need to keep is the removal of `if (!call)`.


Alfred

**Comment by alfredh** [comment]
hi @jmillan 

I made an alternative solution. Could you please test
commit 6d752df5bd6630a19ef82ae22cb6668123c21579
and see if it works for you ?

**Comment by jmillan** [comment]
Hi Alfred,

I'm on vacation right now, almost out of Internet.

I'll check it once I'm back.

Thanks

**Comment by jmillan** [comment]
It's working perfectly @alfredh.

Nice solution!

**Pull request closed without merging**